### PR TITLE
Add Kubeflow application

### LINKED ISSUE #341: A Kubeflow application
This issue is a:
* [ ] Bug Report
* [ ] Enhancement suggestion
* [x] Feature request

## Enhancement suggestions / feature requests

It would be pretty helpful to have a Kubeflow application on the Civo marketplace. Kubeflow is quite popular in the Machine Learning community as can be seen in [kubeflow/kubeflow](https://github.com/kubeflow/kubeflow) and would help the overall Civo community.

> The Kubeflow project is dedicated to making deployments of machine learning (ML) workflows on Kubernetes simple, portable and scalable. Our goal is not to recreate other services, but to provide a straightforward way to deploy best-of-breed open-source systems for ML to diverse infrastructures. Anywhere you are running Kubernetes, you should be able to run Kubeflow.  —[kubeflow.org](https://www.kubeflow.org/)

I would love to contribute this if it sounds like a right fit for the Civo marketplace.

I propose using [Kubeflow Manifests](https://github.com/kubeflow/manifests) to create this application. I tried deploying Kubeflow on a Civo cluster and running some preliminary tests on installing using the Manifests and would propose going forward this way if we go ahead with this issue.

### PR DESCRIPTION
**Pull request by Rishit-dagli**
This PR adds the new Kubeflow application which should make it very easy for users to start using Kubeflow on Civo. I tried to approach this by using `kubectl wait` too however that requires me to separate out the CRDs and namespaces and I thought this approach would just be easier.

If your pull request is to submit a new application to the marketplace, please answer the following questions:

* [x] Have you followed the guidelines in our Contributing document in setting up the application?
* [x] Have you checked that the application your submission is proposing to add does not yet exist on the Marketplace, or as a pull request to be processed?
* [x] Have you tested your application on the Civo managed Kubernetes service? Please include a screenshot.

The Kubernetes Cluster is running:

image

I am able to access the Kubeflow dashboard:

image
image

* [x] Can you confirm that the software you are submitting licensed for use in a publicly-available Kubernetes marketplace?

Yes, please see: https://github.com/kubeflow/kubeflow/blob/master/LICENSE

* [x] Have you included the specific version of the application to be installed?

---

Closes #341

### CODE REVIEW

**Comment by Rishit-dagli** [comment]
@saiyam1814 Did you get the chance to take a look at this PR, I completely understand if you might have been swamped

**Comment by saiyam1814** [comment]
Any updates @Rishit-dagli ?

**Comment by Rishit-dagli** [comment]
> Any updates @Rishit-dagli ?

Sorry, updates for what? Did I miss something?

**Comment by saiyam1814** [comment]
Screenshot 2022-08-17 at 4 53 04 PM
@Rishit-dagli

**Comment by Rishit-dagli** [comment]
Oh yes, I have faced this issue on reviews I have done as well, no one except you can see reviews that show up as "pending", you would need to submit it

Reference: https://github.com/community/community/discussions/10369

**Comment by saiyam1814** [comment]
I did submit

File: kubeflow/install.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+while ! kubectl apply -f https://raw.githubusercontent.com/civo/kubernetes-marketplace/master/kubeflow/kubeflow.yaml; do echo "Retrying to apply resources"; sleep 10; done
```
[the remark this answers is not in the document]
Comment (Rishit-dagli) [reply kubeflow/install.sh:3]: Yep, so this is actually inspired by the [official installation guide](https://github.com/kubeflow/manifests#install-with-a-single-command).

This repeats the command until all the Kubernetes resources in `kubeflow.yaml` are created. If these can't be created the command tries creating the resources again until they are created. In my experiments on Civo, at the end of the second run of `kubectl apply` all the resources are created and the command exits.

**Comment by Rishit-dagli** [comment]
> I did submit

Gotcha, I now see them

File: kubeflow/post_install.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+## Verify Kubeflow deployment
+
+After installation, it will take some time for all Pods to become ready. You should make sure all Pods are ready, to check that all Kubeflow-related Pods are ready, run the following commands:
+
+```sh
+kubectl get pods -n cert-manager
+kubectl get pods -n istio-system
```
[the remark this answers is not in the document]
Comment (Rishit-dagli) [reply kubeflow/post_install.md:7]: Yes this also applies ingress controller and creates a load balancer which should show up as: `<CLUSTER NAME>-istio-system-istio-ingressgateway` on the Civo dashboard

[the remark this answers is not in the document]
Comment (saiyam1814) [reply kubeflow/post_install.md:7]: Ok can you add
required_services:
  - loadbalancer
 in manifest file ?

[the remark this answers is not in the document]
Comment (Rishit-dagli) [reply kubeflow/post_install.md:7]: Thanks for spotting this. I have pushed a commit that fixes this.

**Review by saiyam1814 — APPROVED** [review]
Looks fine

**Comment by Rishit-dagli** [comment]
As a sanity test, I tried installing the application on Civo and it works very well, I even tested individual Kubeflow components!

**Comment by Rishit-dagli** [comment]
@saiyam1814

I was particularly intrigued by how fast the deployment with all components went through, given the [Kubeflow User Survey, page 24](https://15905978862850875460.googlegroups.com/attach/150ca032927e6/'22%20Kubeflow%201.6%20User%20Survey%20-%20initial%20results.pdf?part=0.1.1&view=1&vt=ANaJVrHxiiafRn3qrghxG9Q0cuWgPiJn1g5Ytgx5J-Vzkjd04ghZi3vHUqsIxZFVALRA8IfrU4IiZuN9ABZfVybzcbMwjcDRRIF8YlLtaOdw8bH04hygTeA) shows 30.8% folks have problems installing Kubeflow. Especially paired with ~2 minute cluster creation time of Civo; I can now literally 1 click to deploy the Kubeflow with all components and the latest version ([v1.6.0-rc.1](https://github.com/kubeflow/manifests/releases/tag/v1.6.0-rc.1)) and have it up in <5 minutes (cluster creation + deploying Kubeflow + waiting for all pods to be running)!

To me, the speed (faster than the other 1 click deploys Kubeflow) and 1 click deploy sound a good fit for a blog post which could help many folks. Could I propose this idea for a blog on the [Civo blog](https://www.civo.com/blog) if possible, I would love to put this in a blog for the Civo blog?

**Comment by saiyam1814** [comment]
Yes @Rishit-dagli it works fine but I wanted to make use of a couple of things here one is Civo email for login and autogenerated password. 
we can easily do that.

**Comment by Rishit-dagli** [comment]
Sounds like a great plan, I will start working on this in a while

**Pull request merged**